### ISSUE TITLE
Vertical axis labels

### ISSUE THREAD

**Issue by pchronz** [gampleman/elm-visualization#27]
Hi and thanks for providing elm-visualization!

It would be great to be able to rotate the labels on the axes.
As a concrete use case, I'm trying to build a bar chart based on the bar chart example (https://code.gampleman.eu/elm-visualization/BarChart/).

Currently, the labels overlap when they're too long, which makes them unreadable.
![elm-visualization-overlapping labels](https://user-images.githubusercontent.com/586138/41846995-d4c5b494-7878-11e8-8439-83bdfa860ce8.png)

I've tried to rotate the columns individually in SVG by using the `rotate(90)` transformation in the browser. However, there are two problems with this approach. 

1. The labels are misaligned
2. There doesn't seem to be a way to do this using the visualization API.

![screen shot 2018-06-25 at 1 10 48 pm](https://user-images.githubusercontent.com/586138/41847370-0abd83d2-787a-11e8-9882-d5feb546748e.png)

I've also tried to apply the rotation transformation to the .tick class in the example. However, then the translation of individual columns doesn't work anymore. Furthermore, this approach applies the transformation to the x and y axis. Finally, the problem with alignment would remain.

![screen shot 2018-06-25 at 1 15 50 pm](https://user-images.githubusercontent.com/586138/41847385-16ebd0c8-787a-11e8-8da8-216f0c4edd6b.png)

As a third approach I've limited the size of the labels. In this case some labels turn out to be duplicates, which leads to errors in the rendering:

![screen shot 2018-06-25 at 1 19 01 pm](https://user-images.githubusercontent.com/586138/41847470-6375d7fe-787a-11e8-8f49-fe0fa4f892b0.png)

I'd really appreciate any help on this issue.

Cheers,
Peter

**Comment by gampleman** [comment]
I've got this with modifying the view function in the BarChart example to read like this:

```elm
view : List ( Date, Float ) -> Svg msg
view model =
    svg [ width (toString w ++ "px"), height (toString (h + 10) ++ "px") ]
        [ Svg.style [] [ text """
            .column rect { fill: rgba(118, 214, 78, 0.8); }
            .column text { display: none; }
            .column:hover rect { fill: rgb(118, 214, 78); }
            .column:hover text { display: inline; }

            .x-axis .tick text {
              transform: rotate(45deg);
              transform-origin: -4px 33px;
            }
          """ ]
        , g [ transform ("translate(" ++ toString (padding - 1) ++ ", " ++ toString (h - padding) ++ ")"), class "x-axis" ]
            [ xAxis model ]
        , g [ transform ("translate(" ++ toString (padding - 1) ++ ", " ++ toString padding ++ ")") ]
            [ yAxis ]
        , g [ transform ("translate(" ++ toString padding ++ ", " ++ toString padding ++ ")"), class "series" ] <|
            List.map (column (xScale model)) model
        ]
```

<img width="889" alt="screen shot 2018-06-25 at 12 50 06" src="https://user-images.githubusercontent.com/69144/41848734-6141a6d8-7876-11e8-8744-31e085e3c9dd.png">

Notice the `.x-axis .tick text`: this allows you to target just the text label and not the tick mark. The `transform-origin` allows you to pinpoint where the rotation point will be.

Let me know if this solves the issue for you.

**Comment by pchronz** [comment]
Thank you very much for your help.

Using your suggestions I got the following:

![screen shot 2018-06-25 at 2 56 59 pm](https://user-images.githubusercontent.com/586138/41851466-4521db96-7888-11e8-8a60-c4bcb41a634a.png)

Here I had to fine-tune the `transform-origin` setting and I had to increase the padding in y-direction. As long as the maximum length of the labels is fixed this approach seems to be okay. A better approach in the future might be to generate the text in the origin of the SVG group and then to rotate and translate. I imagine that would make it possible to align all labels nicely. Though I have no idea how to do that in SVG.

So, for my purposes the solution is good enough. Feel free to close the issue.
